src/main/java/com/google/cloud/genomics/dataflow/utils/VariantUtils.java

```diff
@@ -28,7 +28,10 @@
 
   public static boolean isVariant(Variant variant) {
     List<String> alternateBases = variant.getAlternateBases();
-    return !(null == alternateBases || alternateBases.isEmpty());
+    // The same deletion can be specified as [CAG -> C] or [AG -> null], so double check that the
+    // reference bases are also of length 1 when there are no alternates.
+    return !(LENGTH_IS_1.apply(variant.getReferenceBases()) 
+        && (null == alternateBases || alternateBases.isEmpty()));
```

Comment:
@jean-philippe-martin is right.  @deflaux, the easiest thing is to implement exactly the ideas from the VCF specifications for `REF` and `ALT`:

*4) REF - reference base(s): Each base must be one of A,C,G,T,N (case insensitive). Multiple bases are permitted. The value in the POS field refers to the position of the first base in the String. For simple insertions and deletions in which either the REF or one of the ALT alleles would otherwise be null/empty, the REF and ALT Strings must include the base before the event (which must be reflected in the POS field), unless the event occurs at position 1 on the contig in which case it must include the base after the event; this padding base is not required (although it is permitted) for e.g. complex substitutions or other events where all alleles have at least one base represented in their Strings. If any of the ALT alleles is a symbolic allele (an angle-bracketed ID String “\<ID\>”) then the padding base is required and POS denotes the coordinate of the base preceding the polymorphism. Tools processing VCF files are not required to preserve case in the allele Strings. (String, Required).*

_5) ALT - alternate base(s): Comma separated list of alternate non-reference alleles called on at least one of the samples. Options are base Strings made up of the bases A,C,G,T,N,\*, (case insensitive) or an angle-bracketed ID String (“\<ID\>”) or a breakend replacement string as described in the section on breakends. The ‘\*’ allele is reserved to indicate that the allele is missing due to a upstream deletion. If there are no alternative alleles, then the missing value should be used. Tools processing VCF files are not required to preserve case in the allele String, except for IDs, which are case sensitive. (String; no whitespace, commas, or angle-brackets are permitted in the ID String itself)_

Below is a link to the Variant Call Format (VCF) Version 4.2 Specification document:

http://samtools.github.io/hts-specs/VCFv4.2.pdf

Hope it helps,
~p